### ISSUE TITLE
After upgrading to Blazor latest Released version, Not able to publish to Azure App

### ISSUE THREAD

**Issue by nssidhu** [dotnet/aspnetcore#22064]
Publishing via  Microsoft Visual Studio Enterprise 2019 Preview
Version 16.6.0 Preview 6.0
After upgrading to released version of Blazor, i am no longer able to publish the app to Azure Appservice.

running into the following Error, Works fine on local development box.
=============================================================
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.Publish.Framework.ViewModel.ProfileSelectorViewModel.ProfileStateUpdater.Dispose()
   at Microsoft.Publish.Framework.ViewModel.ProfileSelectorViewModel.set_SelectedProfile(IProfileVisual value)
   at Microsoft.Publish.Framework.ViewModel.ProfileSelectorViewModel.<ReloadProfilesAsync>d__160.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.WebTools.Azure.Publish.PublishProviders.AppServicePublishProfileVisual.<AspNetCoreInWindowsAppServicePostPublishAsync>d__58.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.WebTools.Azure.Publish.PublishProviders.AppServicePublishProfileVisual.<PostPublishAsync>d__51.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Publish.Framework.ViewModel.ProfileSelectorViewModel.<RunPublishTaskAsync>d__169.MoveNext()

**Comment by davidfowl** [comment]
cc @vijayrkn

**Comment by vijayrkn** [comment]
@abpiskunov can you please take a look?

**Comment by abpiskunov** [comment]
@nssidhu could you please specify exact repro steps? Are you opening existing publish profile and trying to publish? Or you creating a new publish profile and trying to publish right away after it was created? Anything else?

**Comment by nssidhu** [comment]
First i published using the exisiting publish profile.
next i created new Azure App Service plan and new publish profile , but it did not help.

I am able to reproduce it on local dev machine if i build in release mode, i encounter the problem.
In the debug mode everything works fine.

I can provide the sample app privately if that would help.

**Comment by abpiskunov** [comment]
I see, i found a problem that could lead to this exception in code. If you share the app for testing it - it would be awesome. How do you want to share it?

**Comment by nssidhu** [comment]
let me knwo the email address where i can send. 
Just for clarification i am talking about Blazor Webassembly app.

**Comment by abpiskunov** [comment]
Please send it at: antonpis@microsoft.com

**Comment by nssidhu** [comment]
sent email witth link to ondrive, file was too big to be sent over email.
let me know if you can open the file

**Comment by abpiskunov** [comment]
@nssidhu i downloaded zip , thanks. About the second problem you mentioned in the email , please open a separate github issue here it is unrelated to this issue.

**Comment by mkArtakMSFT** [comment]
Closing as there is no pending action from our team on this and the other issues are tracked separately.

**Comment by nssidhu** [comment]
hmmm, i think i am still waiting on the resolution on the main reported problem. is it closed mistakenly ?

**Comment by abpiskunov** [comment]
It belongs to different team (VS WebTools) , we track it in separate internal repo. We are working on the fix and it will be resolved on coming release.

**Comment by nssidhu** [comment]
ok.

**Comment by nssidhu** [comment]
I was able to narrow it down to the following file.
  <PackageReference Include="Microsoft.AspNetCore.Components.Authorization" Version="3.1.4" />

**Comment by nssidhu** [comment]
I was able to fix it by adding the following in the blazor client project file
 <BlazorWebAssemblyEnableLinking>false</BlazorWebAssemblyEnableLinking>
Not sure if that is a temporary fix, but would like to hear from the team if that is a viable fix.

**Comment by abpiskunov** [comment]
You mean publish worked after you made that change?

**Comment by nssidhu** [comment]
So i was able to reproduce it locally as well. In debug mode on my local system everything worked fine.
When i changed it to release mode i started seeing the same error that i would get during publish.
after making the aforementioned change i was able to run the blazor app locally in release mode as well able to publish .

but unfortunatley the published app still fails at runtime with the same error, so i would still considere it as unresolved

**Comment by abpiskunov** [comment]
Hm i am a little confused now. The error stack that you pasted on the very top comes from VisualStudio publish view and can happen when yu click Publish button or switch profiles selection. It has nothing to do with runtime behavior of the app. What is the runtime error you are having? 

This current github issue is focusing only on the stack above, if you have different problem that is runtime related you should open a separate issue in this repo.

**Comment by nssidhu** [comment]
Well the issue was that publish was failing, on further anlyzing the problem i found that underlying problem of publish failing was due to code not compiling when set to release mode.Further anlysis led me to the packge that was throwing an error.
I think you can play around with the sample app that i provided.
So yes the initial issue seemed to be publishing error, but the root cause was that code was failing to compile in release mode. the code works fin in debug mode.
it is just my anaylsis that i am finding the underlying problem.================================================================================================================================================
    On Thursday, May 21, 2020, 01:46:59 PM CDT, Anton Piskunov <notifications@github.com> wrote:  
 
 


Hm i am a little confused now. The error stack that you pasted on the very top comes from VisualStudio publish view and can happen when yu click Publish button or switch profiles selection. It has nothing to do with runtime behavior of the app. What is the runtime error you are having?

This current github issue is focusing only on the stack above, if you have different problem that is runtime related you should open a separate issue in this repo.

**Comment by abpiskunov** [comment]
I see, then you probably need to rename this issue and add a runtime error with all details and then re-open issue (or create a new one). VS issue that was on top we still will fix, but it seems unrelated to your actual issue.

**Comment by nssidhu** [comment]
i opened separate issue for the underlying errors.
